tests/sqllogictest/cast.slt

```diff
@@ -104,4 +104,14 @@ SELECT unnest('{"[1,2,3]", "[4,5,6]"}'::vector[]);
 query I
 SELECT '{"[1,2,3]"}'::vector[];
 ----
-{"[1, 2, 3]"}
\ No newline at end of file
+{"[1, 2, 3]"}
+
+# vector to vecf16
+query I
+SELECT '[1,]'::vector::vecf16;
+----
+[1]
+
+# vecf16 to vector should fail
```

Comment:
We may provide a cast from `vecf16` to `vector` for convenience. Could you remove it?

Reply:
I personally like codifying negative cases, but its up to you.  

The reason why, at least initially, to do this is because I think going from vecf16 to vector might be a bad idea because you already lost that precision.

Reply:
Actually, I'll remove this and make the cast explicit.